lib/APPReinforcementLearning/src/App.cpp

```diff
@@ -123,35 +125,56 @@ void App::loop()
             payload = {SMPChannelPayload::Status::DONE};
         }
 
-        /* Ignoring return value, as error handling is not available. */
-        (void)m_smpServer.sendData(m_serialMuxProtChannelIdStatus, &payload, sizeof(payload));
+        isDataSent = m_smpServer.sendData(m_serialMuxProtChannelIdStatus, &payload, sizeof(payload));
```

Comment:
This is too dificult to read. Why not try something like this?
```suggestion
        if (false == m_smpServer.sendData(m_serialMuxProtChannelIdStatus, &payload, sizeof(payload)))
{
    /* Failed to send Status data to the Supervisor. Go to error state. */
    ErrorState::getInstance().setErrorMsg("SD_SF");
    m_systemStateMachine.setState(&ErrorState::getInstance());
}
else
{
    m_statusTimer.restart();
}
```